pkg/agent/datapath/multiBridgeDatapath.go

```diff
@@ -173,6 +179,49 @@ type Bridge interface {
 	MultipartReply(sw *ofctrl.OFSwitch, rep *openflow13.MultipartReply)
 }
 
+type PacketInHandler interface {
+	HandlePacketIn(packetIn *ofctrl.PacketIn)
+}
+
+type PacketInHandlerFuncs struct {
+	PacketInHandlerFunc func(packetIn *ofctrl.PacketIn)
+}
+
+func (handler PacketInHandlerFuncs) HandlePacketIn(packetIn *ofctrl.PacketIn) {
+	if handler.PacketInHandlerFunc != nil {
+		handler.PacketInHandlerFunc(packetIn)
+	}
+}
+
+func (datapathManager *DpManager) RegisterPacketInHandler(handler PacketInHandler) {
+	if handler == nil {
+		log.Fatalf("Failed to register packetInHandler: register nil PacketInHandler is not allow")
+	}
+	if datapathManager.PacketInHandler != nil {
+		log.Fatalf("Failed to register PacketInHandler: datapathManager PacketInHandler already register")
+	}
+	datapathManager.PacketInHandler = handler
+}
+
+type Packet struct {
```

Comment:
Do we support the VLAN tag during the probe packet generate?

Reply:
Temporarily not supported